.github/workflows/golangci-lint.yml

```diff
@@ -27,7 +27,7 @@ jobs:
       - name: golangci-lint
         uses: golangci/golangci-lint-action@v3
         with:
-          version: v1.51.0
+          version: v1.52.0
```

Comment:
The version was pinned in https://github.com/hashicorp/terraform-provider-aws/pull/29272.
We may want to revert to `latest`?

Reply:
My plan is to make it use the version set in `.ci/tools/go.mod`, so that the same version can be used locally and in GH

Reply:
👍